core/p2pCore.go

```diff
@@ -63,3 +63,22 @@ type QueryP2PPeerInfo struct {
 	PeerSubType   string   `json:"peersubtype"`
 	Addresses     []string `json:"addresses"`
 }
+
+// PeerTopicType represents the type of a peer in regards to the topic it is used
+type PeerTopicType string
+
+// String returns a string version of the peer topic type
```

Comment:
no need for redefining the String method for a type derived from string

Reply:
I will leave it as it is. A `.String()` call looks better than a cast.